src/main/webapp/go.js

```diff
@@ -436,31 +406,20 @@ function checkUserDestinationGuess() { //eslint-disable-line
 
   // Check to see if userGuess can be used to identify the correct place
   // using the Places library.
-  checkGuessWithPlaceIDs(userGuess);
-}
-
-/**
- * Retrieves place ID for the answer, and compares this to the place
- * ID retrieved for the user's guess.
- * @param {String} userGuess User's guess for current destination.
- */
-function checkGuessWithPlaceIDs(userGuess) {
-  const answerRequest = createRequestForPlaceID(hunt.getCurDestName());
-  service.findPlaceFromQuery(answerRequest, function(results, status) {
-    if (status === google.maps.places.PlacesServiceStatus.OK) {
-      const answerID = results[0].place_id;
-      getPlaceIDOfGuess(userGuess, answerID);
-    }
-  });
+  checkGuessWithIDOrEntities(userGuess, hunt.getCurDestId());
 }
 
 /**
+ * Compares the Place ID retrieved using {@code userGuess} with
+ * {@code answerID}. If the IDs match, the user's guess is correct.
+ * If the IDs don't match or if an ID could not be retrieved,
+ * then entity extraction is performed.
  * @param {String} userGuess The user's guess for the name of the
  * current destination.
  * @param {String} answerID Place ID corresponding to the correct
  * destination
  */
-function getPlaceIDOfGuess(userGuess, answerID) {
+function checkGuessWithIDOrEntities(userGuess, answerID) {
   const userRequest = createRequestForPlaceID(userGuess);
   service.findPlaceFromQuery(userRequest, function(results, status) {
     if (status === google.maps.places.PlacesServiceStatus.OK) {
```

Comment:
To reduce redundant code (`checkGuessWithEntities(userGuess)`), I suggest making this line
```suggestion
    if (status === google.maps.places.PlacesServiceStatus.OK && results[0].place_id === answerID) {
```
Then you can simplify down to a single `if` branch and single `else` branch.